x-pack/plugins/index_lifecycle_management/server/routes/api/indices/register_get_affected_route.js

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { callWithRequestFactory } from '../../../lib/call_with_request_factory';
+import { isEsErrorFactory } from '../../../lib/is_es_error_factory';
+import { wrapEsError, wrapUnknownError } from '../../../lib/error_wrappers';
+import { licensePreRoutingFactory } from '../../../lib/license_pre_routing_factory';
+
+async function fetchTemplates(callWithRequest) {
+  const params = {
+    method: 'GET',
+    path: '/_template',
+    // we allow 404 in case there are no templates
+    ignore: [404]
+  };
+
+  return await callWithRequest('transport.request', params);
+}
+
+async function getAffectedIndices(
+  callWithRequest,
+  indexTemplateName,
+  policyName
+) {
+  const templates = await fetchTemplates(callWithRequest);
+
+  if (!templates || Object.keys(templates).length === 0 | templates.status === 404) {
```

Comment:
We don't want the call to throw an error on 404 because the user may just legitimately not have any templates, which will return a 404.  This is not an error, but rather an expected case.